qa/acceptance-tests/src/test/java/io/camunda/it/client/RoleIntegrationTest.java

```diff
@@ -104,11 +117,184 @@ void shouldRejectCreationIfMissingRoleName() {
+        .resourceId("resourceId")
+        .resourceType(ResourceType.RESOURCE)
+        .permissionTypes(PermissionType.CREATE, PermissionType.READ)
+        .send()
+        .join();
+
+    // Verify it was created
+    Awaitility.await()
+        .untilAsserted(
+            () ->
+                assertThat(
+                        searchAuthorizations(
+                                camundaClient.getConfiguration().getRestAddress().toString(),
+                                "admin")
+                            .items())
+                    .anyMatch(
+                        auth ->
+                            auth.resourceId().equals("resourceId")
+                                && auth.resourceType().equals(ResourceType.RESOURCE)
+                                && auth.ownerId().equals(ROLE_ID_4)));
+
+    camundaClient.newDeleteRoleCommand(ROLE_ID_4).send().join();
+
+    // then
+    Awaitility.await("Role is deleted")
+        .untilAsserted(
+            () ->
+                assertThatThrownBy(() -> camundaClient.newRoleGetRequest(ROLE_ID_4).send().join())
+                    .isInstanceOf(ProblemException.class)
+                    .hasMessageContaining("Failed with code 404: 'Not Found'"));
+
+    Awaitility.await("Authorization is deleted")
+        .untilAsserted(
+            () ->
+                assertThat(
+                        searchAuthorizations(
+                                camundaClient.getConfiguration().getRestAddress().toString(),
+                                "admin")
+                            .items())
+                    .noneMatch(
+                        auth ->
+                            auth.resourceId().equals("resourceId")
+                                && auth.resourceType().equals(ResourceType.RESOURCE)
+                                && auth.ownerId().equals(ROLE_ID_4)));
+  }
+
+  private static void assertRoleCreated(
+      final String roleId, final String roleName, final String description) {
+    Awaitility.await("Role is created and exported")
+        .ignoreExceptionsInstanceOf(ProblemException.class)
+        .untilAsserted(
+            () -> {
+              final var role = camundaClient.newRoleGetRequest(roleId).send().join();
+              assertThat(role).isNotNull();
+              assertThat(role.getRoleId()).isEqualTo(roleId);
+              assertThat(role.getName()).isEqualTo(roleName);
+              assertThat(role.getDescription()).isEqualTo(description);
+              assertThat(role.getRoleKey()).isPositive();
+            });
+  }
+
+  // TODO once available, this test should use the client to make the request
+  private static AuthorizationSearchResponse searchAuthorizations(
+      final String restAddress, final String username)
+      throws URISyntaxException, IOException, InterruptedException {
+    final var encodedCredentials =
+        Base64.getEncoder().encodeToString("%s:%s".formatted(username, "password").getBytes());
```

Comment:
🔧 Authentication is disabled in this test, so we don't have to btoher with credentials. You can remov this and and the `.header` call on the request.